orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/UpsertIntentTask.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.keel.model.Intent
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentDryRunResponse
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentRequest
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentResponse
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+
+@Component
+class UpsertIntentTask
+@Autowired constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage) : TaskResult {
+    if (!stage.context.containsKey("intents")) {
+      throw IllegalArgumentException("Missing required task parameter (intents)")
+    }
+
+    if (!stage.context.containsKey("dryRun")) {
+      throw IllegalArgumentException("Missing required task parameter (dryRun)")
+    }
+
+    val upsertIntentRequest = UpsertIntentRequest(
+      intents = keelObjectMapper.convertValue(stage.context["intents"], object : TypeReference<List<Intent>>() {}),
+      dryRun = java.lang.Boolean.valueOf(stage.context["dryRun"].toString())!!
```

Comment:
Can we not do `stage.context["dryRun"].toString().toBoolean()` instead?

Reply:
I also think avoiding `!!` is a good idea. That's a sure-fire way to get an exception, especially with orca contexts. What we could do is something like... `if (stage.context["dryRun"] == null) false else stage.context["dryRun"].toString().toBoolean()` so we're handling the null case explicitly.

Reply:
Yes, that was leftover from some intellij magic of pasting java to kotlin. Fixing.